src/Microsoft.IdentityModel.Protocol.Extensions/SamlSecurityTokenHandler.cs

```diff
@@ -287,118 +290,160 @@ public virtual ClaimsPrincipal ValidateToken(string securityToken, TokenValidati
                 throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, ErrorMessages.IDX10209, securityToken.Length, MaximumTokenSizeInBytes));
             }
 
+            SecurityKeyResolver issuerTokenResolver = new SecurityKeyResolver(securityToken, validationParameters);
+
             // Calling System.IdentityModel.Tokens.SamlSecurityTokenHandler requires Configuration and IssuerTokenResolver be set.
             Configuration = new SecurityTokenHandlerConfiguration
             {
-                IssuerTokenResolver = new SecurityKeyResolver(securityToken, validationParameters),
+                IssuerTokenResolver = issuerTokenResolver,
                 MaxClockSkew = validationParameters.ClockSkew,
             };
 
-            SamlSecurityToken samlToken;
-            using (StringReader sr = new StringReader(securityToken))
+            try
             {
-                using (XmlDictionaryReader reader = XmlDictionaryReader.CreateDictionaryReader(XmlReader.Create(sr)))
+                SamlSecurityToken samlToken;
+                using (StringReader sr = new StringReader(securityToken))
                 {
-                    samlToken = ReadToken(reader, validationParameters) as SamlSecurityToken;
+                    using (XmlDictionaryReader reader = XmlDictionaryReader.CreateDictionaryReader(XmlReader.Create(sr)))
+                    {
+                        samlToken = ReadToken(reader, validationParameters) as SamlSecurityToken;
+                    }
                 }
-            }
 
-            if (samlToken.Assertion == null)
-            {
-                throw new ArgumentException(ErrorMessages.IDX10202);
-            }
+                if (samlToken.Assertion == null)
```

Comment:
We should check if samlToken == null. It is unlikely, but samlToken.Assertion could null ref. This error was already here.